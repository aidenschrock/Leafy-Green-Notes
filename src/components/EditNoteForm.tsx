import { useEffect, useState } from 'react';
import Modal from '@leafygreen-ui/modal';
import TextInput from '@leafygreen-ui/text-input';
import TextArea from '@leafygreen-ui/text-area';
import Button from '@leafygreen-ui/button';

interface FuncProps {
    handleNoteSave({ title, content }): void;
    handleNoteDelete(): void;
    open: boolean;
    setOpen: any;
    data: any;
}

const EditNoteForm: React.FC<FuncProps> = (props) => {
    const [title, setTitle] = useState(props.data?.title ?? '');
    const [content, setContent] = useState(props.data?.content ?? '');

    useEffect(() => {
        setTitle(props.data?.title)
        setContent(props.data?.content)
    }, [props.data?.title, props.data?.content])

    return (
        <Modal className="edit-note" open={props.open} setOpen={props.setOpen}>
            <TextInput
                data-testid="title-input"
                name="note-title"
                className="input"
                label="Note Title"
                placeholder="My Note"
                value={title}
                onChange={event => {
                    setTitle(event.target.value)
                }}
            />
            <TextArea
                data-testid="content-input"
                name="note-content"
                className="input"
                label="Note Content"
                placeholder="Reminder: Eat more vegetables!"
                value={content}
                onChange={event => {
                    setContent(event.target.value)
                }}
            />
            <Button id="delete-button" onClick={() => props.handleNoteDelete()} className="delete-button" variant={'danger'}>Delete</Button>
            <Button id="save-button" onClick={() => props.handleNoteSave({
                title: title,
                content: content
            })} variant={'primary'} className="save-button"> Save </Button>
        </Modal>
    )
}



export default EditNoteForm;