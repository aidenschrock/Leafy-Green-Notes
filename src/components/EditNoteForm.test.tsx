import { render, screen, fireEvent } from '@testing-library/react';
import EditNoteForm from './EditNoteForm';
import { shallow } from 'enzyme';
import userEvent from '@testing-library/user-event'


  describe('EditNoteForm', () => {
    it('should render correctly', () => {
      const component = shallow(<EditNoteForm  open={true} setOpen={null} data={null} handleNoteSave={null} handleNoteDelete={null} />);
      expect(component).toMatchSnapshot();
    });
  });

  test('title type', async () => {
    render(<EditNoteForm  open={true} setOpen={null} data={null} handleNoteSave={null} handleNoteDelete={null}/>)
  const textbox = screen.getByTestId('title-input');
  userEvent.type(textbox, 'Title');
  expect(await textbox).toHaveValue('Title');
  });

  test('content type', async () => {
    render(<EditNoteForm  open={true} setOpen={null} data={null} handleNoteSave={null} handleNoteDelete={null}/>)
  const textbox = screen.getByTestId('content-input');
  userEvent.type(textbox, 'content');
  expect(await textbox).toHaveValue('content');
  });

  it('submits when clicked', () => {
    const mockSave = jest.fn();
    const mockDelete = jest.fn();
    const wrapper = shallow(<EditNoteForm open={true} setOpen={null} data={null} handleNoteSave={mockSave} handleNoteDelete={mockDelete}/>);
    const saveButtonElement  = wrapper.find('#save-button');
    const deleteButtonElement  = wrapper.find('#delete-button');
    saveButtonElement.simulate('click');
    deleteButtonElement.simulate('click');
    expect(mockSave.mock.calls.length).toEqual(1);
    expect(mockDelete.mock.calls.length).toEqual(1);
  });

  it('should send data when note is saved', () => {
    const handleSubmit = jest.fn();
    render(<EditNoteForm open={true} setOpen={null} data={null} handleNoteSave={handleSubmit} handleNoteDelete={null} />)
    const titleInput = screen.getByTestId('title-input');
    const contentInput = screen.getByTestId('content-input');

    fireEvent.change(titleInput, {target: {value: "Note Title"}}) 
    fireEvent.change(contentInput, {target: {value: "contents"}}) 

    const buttonElement  = screen.getByText(/Save/i)
 
    fireEvent.click(buttonElement)
    expect(handleSubmit).toHaveBeenCalledWith({"content":'contents',"title":"Note Title"}) 
  
  });
  