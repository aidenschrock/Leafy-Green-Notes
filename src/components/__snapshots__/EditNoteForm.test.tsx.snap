// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNoteForm should render correctly 1`] = `
<Modal
  className="edit-note"
  open={true}
  setOpen={null}
>
  <TextInput
    className="input"
    data-testid="title-input"
    label="Note Title"
    name="note-title"
    onChange={[Function]}
    placeholder="My Note"
    value=""
  />
  <TextArea
    className="input"
    data-testid="content-input"
    label="Note Content"
    name="note-content"
    onChange={[Function]}
    placeholder="Reminder: Eat more vegetables!"
    value=""
  />
  <Button
    className="delete-button"
    id="delete-button"
    onClick={[Function]}
    variant="danger"
  >
    Delete
  </Button>
  <Button
    className="save-button"
    id="save-button"
    onClick={[Function]}
    variant="primary"
  >
     Save 
  </Button>
</Modal>
`;
